nilearn/plotting/surf_plotting.py

```diff
@@ -422,6 +426,248 @@ def plot_surf_stat_map(surf_mesh, stat_map, bg_map=None,
+    if invalid_hemi:
+        supported = "Supported hemispheres:\n" + str(VALID_HEMISPHERES)
+        raise ValueError("Invalid hemispheres definition!\n" + supported)
+    return hemispheres
+
+
+def _check_views(views) -> list:
+    if not isinstance(views, list):
+        views = [views]
+    invalid_view = any([view not in VALID_VIEWS for view in views])
+    if invalid_view:
+        supported = "Supported views:\n" + str(VALID_VIEWS)
+        raise ValueError("Invalid view definition!\n" + supported)
+    return views
+
+
+def _colorbar_from_array(array, vmax,
+                         threshold, symmetric_cbar,
+                         kwargs,
+                         cmap='cold_hot'):
+    """ Generate a custom colorbar for an array.
+
+    Internal function used by plot_img_on_surf
+
+    array: Any 3D array
+
+    vmax : float, optional (default=None)
+        upper bound for plotting of stat_map values.
+
+    threshold : float, optional (default=None)
+        If None is given, the colorbar is not thresholded.
+        If a number is given, it is used to threshold the colorbar.
+        Absolute values lower than threshold are shown in gray.
+
+    symmetric_cbar : bool or 'auto', optional, default 'auto'
+         Specifies whether the colorbar should range from -vmax to vmax
+         or from vmin to vmax. Setting to 'auto' will select the latter
+         if the range of the whole image is either positive or negative.
+         Note: The colormap will always range from -vmax to vmax.
+
+    kwargs: extra arguments passed to _get_colorbar_and_data_ranges
+
+    cmap: str, optional (default='cold_hot')
+        the name of a matplotlib or nilearn colormap.
+    """
+
+    cbar_vmin, cbar_vmax, vmin, vmax = _get_colorbar_and_data_ranges(
+        array, vmax, symmetric_cbar, kwargs)
+
+    norm = Normalize(vmin=vmin, vmax=vmax)
+    cmaplist = [cmap(i) for i in range(cmap.N)]
+
+    if threshold is None:
+        threshold = 0.
+
+    # set colors to grey for absolute values < threshold
+    istart = int(norm(-threshold, clip=True) * (cmap.N - 1))
+    istop = int(norm(threshold, clip=True) * (cmap.N - 1))
+    for i in range(istart, istop):
+        cmaplist[i] = (0.5, 0.5, 0.5, 1.)
+    our_cmap = LinearSegmentedColormap.from_list('Custom cmap',
+                                                 cmaplist, cmap.N)
+    sm = plt.cm.ScalarMappable(cmap=our_cmap,
+                               norm=plt.Normalize(vmin=vmin, vmax=vmax))
+    # fake up the array of the scalar mappable.
+    sm._A = []
+
+    return sm
+
+
+def plot_img_on_surf(stat_map, surf_mesh=None, mask_img=None,
+                     hemispheres=['left', 'right'],
+                     inflate=False,
+                     views=['lateral', 'medial'],
+                     output_file=None, title=None, colorbar=True,
+                     vmax=None, threshold=None, symmetric_cbar='auto',
+                     cmap='cold_hot', aspect_ratio=2.75, **kwargs):
```

Comment:
I would suggest `aspect_ratio=1.4` then multiply it by `len(hemispheres)` at the moment the brains are twice as wide when plotting only one hemisphere

Reply:
Done.